File: frontend/src/types/dataImport.ts
export type ImportType="PRODUCTS"|"CUSTOMERS"|"SALES";
export interface ImportError {row_number:number;row_data:Record<string,string>;error_type:string;message:string}
export interface ImportJob {id:number;import_type:ImportType;filename:string;status:string;total_records:number;valid_records:number;successful_records:number;failed_records:number;duplicate_records:number;columns:string[];preview:Record<string,string>[];errors:ImportError[];created_at:string|null;completed_at:string|null;uploaded_by_name:string|null}

File: backend/app/schemas/data_import.py
from datetime import datetime
from typing import Any
from pydantic import BaseModel

class ImportErrorOut(BaseModel):
    row_number: int
    row_data: dict[str, Any]
    error_type: str
    message: str

class ImportOut(BaseModel):
    id: int
    import_type: str
    filename: str
    status: str
    total_records: int
    valid_records: int
    successful_records: int
    failed_records: int
    duplicate_records: int
    columns: list[str] = []
    preview: list[dict[str, Any]] = []
    errors: list[ImportErrorOut] = []
    created_at: datetime | None = None
    completed_at: datetime | None = None
    uploaded_by_name: str | None = None
